src/routes/tools/local-climate-change-snapshot/index.svelte

```diff
@@ -34,12 +34,86 @@
 </script>
 
 <script>
+  import { goto, stores as sapperStores } from "@sapper/app";
+  import { onMount } from "svelte";
+  import { VALID_BOUNDARY_TYPES } from "./_constants";
+  import { isValidNumber } from "./_helpers";
+  import { locationStore } from "~/routes/tools/_common/stores";
+  import { serialize } from "~/helpers/utilities";
   import { Resources } from "~/components/tools/Partials";
+  import { setInitialLocation } from "~/routes/tools/_common/helpers";
   import { Header } from "./_common";
   import SelectLocation from "./_select-location/SelectLocation.svelte";
 
   export let tool;
   export let resources;
+
+  const { page } = sapperStores();
+  const { location, boundary } = locationStore;
+
+  let selectedLocation = null;
+  let searchValue = "";
+  let selectedRadio = "locagrid";
+
+  onMount(() => {
+    cleanUpPrevDom();
+    initApp();
+  });
+
+  async function initApp() {
+    let loc;
+    let {
+      query: { lat, lng, boundary: boundaryType },
+    } = $page;
+    lat = +lat;
+    lng = +lng;
+    if (
+      isValidNumber(lat) &&
+      isValidNumber(lng) &&
+      VALID_BOUNDARY_TYPES.has(boundaryType)
+    ) {
+      try {
+        loc = await setInitialLocation(lng, lat, boundaryType);
```

Comment:
Would not want to fallback to the default location here, it should be blank for users when they first arrive at the page.